presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+            commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+            Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                    icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+            AppendFiles appendFiles = transaction.newFastAppend();
+            for (CommitTaskData task : commitTasks) {
+                DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
+                        .withInputFile(HadoopInputFile.fromLocation(task.getPath(), getConfiguration(session, table.getSchemaName())))
+                        .withFormat(table.getFileFormat())
+                        .withMetrics(task.getMetrics().metrics());
+
+                if (!icebergTable.spec().fields().isEmpty()) {
+                    String partitionDataJson = task.getPartitionDataJson()
+                            .orElseThrow(() -> new VerifyException("No partition data for partitioned table"));
+                    builder.withPartition(PartitionData.fromJson(partitionDataJson, partitionColumnTypes));
+                }
+
+                appendFiles.appendFile(builder.build());
+            }
+
+            // Update the map that tracks tables on which the materialized view depends and the corresponding snapshot ids of the table
+            String tableSnapshotIDString = Joiner.on(",").withKeyValueSeparator("=").join(transactionHandle.getTableSnapshotIdMap());
+            appendFiles.set(DEPENDS_ON_TABLES, tableSnapshotIDString);
+            appendFiles.commit();
+        }
+        transaction.commitTransaction();
+        return Optional.of(new HiveWrittenPartitions(commitTasks.stream()
+            .map(CommitTaskData::getPath)
+            .collect(toImmutableList())));
+    }
+
+    private boolean isAMaterializedView(Table table)
+    {
+        if (table.getTableType().equals(org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW.name()) &&
+                "true".equals(table.getParameters().get(PRESTO_VIEW_FLAG)) &&
+                table.getParameters().containsKey(STORAGE_TABLE)) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isAMaterializedView(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        final HiveIdentity identity = new HiveIdentity(session);
+        if (metastore.getTable(identity, schemaTableName.getSchemaName(), schemaTableName.getTableName()).isPresent()) {
+            Table table = metastore.getTable(identity, schemaTableName.getSchemaName(), schemaTableName.getTableName()).get();
+            return isAMaterializedView(table);
+        }
+        return false;
+    }
+
+    public static List<ConnectorMaterializedViewDefinition.Column> icebergSchemaToMaterializedViewCols(TypeManager typeManager,
```

Comment:
Nit: wrap all arguments or none

Reply:
This method looks unused. Removed it. Point noted around wrapping all arguments or null.